crates/sparrow-backend/src/logical_to_physical.rs

```diff
@@ -260,20 +319,21 @@ impl LogicalToPhysical {
             // trivially instantiated within a projection, but we do this check here
             // for the case of the expression being the result of a non-projection
             // step, since it lets us avoid adding the projection step.
-            result.step_id
+            result_step_id
         } else {
             // Add a projection for the remaining expressions.
             //
             // We could attempt to add them to the step if it is already a projection, but we'd
             // need to deal with mutating the expressions in the projection. For now, we think it
             // is easier to treat steps as immutable once created, and then combine steps in a
             // later optimization pass of the physical plan.
+            let input_type = self.plan.step_result_type(result_step_id).clone();
```

Comment:
Yeah -- here the `input_type` is the "input to the projection" which is applied to the `result_step`. Therefore, the `input_type` is the result type of the result step. What clarification would you suggest? Where could it be elaborated?

Reply:
Ah, I think it was confusing to me because I didn't understand the meaning of "Add a projection for the remaining expressions". A result step may still require execution of addition expressions to produce the output. So the confusion was around seeing the `result` being found, but that potentially not being the actual output of the query. 

The type vars make sense now. As does the logic here. I think it's clear as is now.